packages/neuron-wallet/src/controllers/api.ts

```diff
@@ -158,6 +156,11 @@ export default class ApiController {
     return WalletsController.getAllAddresses(id)
   }
 
+  @MapApiResponse
+  public static async requestPassword({ walletID, action }: { walletID: string, action: 'delete-wallet' | 'backup-wallet' }) {
```

Comment:
If I understand this correctly this doesn't need the backend to do any actual work, except dispatching `delete-wallet` or `backup-wallet` action to the UI? If so why not just remove this passing around and call UI action directly?

Reply:
Reason to keep this: Menu triggered action would go through this API from background as well.